docs/remote_debugging.md

```diff
@@ -0,0 +1,258 @@
+
+## Quick start with VSCode or Chrome DevTools
+
+If you use VSCode or Chrome DevTools **and** are running the program and debugger in the same environment (e.g. on the same machine without using container), `--open` flag can do both steps 1 and 2 for you.
+
+For example:
+
+```shell
+$ rdbg --open=vscode target.rb
+# chrome users can specify --open=chrome instead
+```
+
+It will open a debug port, launch VSCode, and attach to it automatically.
+
+# Invoke program as a remote debuggee
+
+There are multiple ways to run your program as a debuggee:
+
+Stop at program start | [`rdbg` option](#rdbg---open-or-rdbg--o-for-short) | [require](#require-debugopen) | [debugger API](#debuggeropen)
+---|---|---|---|
+Yes | `rdbg --open` | `require "debug/open"` | `DEBUGGER__.open`
+No | `rdbg --open --nonstop` | `require "debug/open_nonstop"` | `DEBUGGER__.open(nonstop: true)`
+
+## `rdbg --open` (or `rdbg -O` for short)
+
+When `--open` flag is used without specifying a client, it'll start the program as remote debuggee and wait for the connection:
+
+```shell
+$ rdbg --open target.rb
+DEBUGGER: Session start (pid: 7773)
+DEBUGGER: Debugger can attach via UNIX domain socket (/home/ko1/.ruby-debug-sock/ruby-debug-ko1-7773)
+DEBUGGER: wait for debugger connection...
+```
+
+By default, `rdbg --open` uses UNIX domain socket and generates path name automatically (`/home/ko1/.ruby-debug-sock/ruby-debug-ko1-7773` in this case).
+
+## `require 'debug/open'`
+
+If you can modify the program, you can open a debugging port by adding `require 'debug/open'` to the program.
+
+If you don't want to stop the program when `require 'debug/open'` is executed, you can instead use `require 'debug/open_nonstop'`.
+
+> **Note**
+> Please do not leave these requires in your program as they will allow other people to connect to it.
+
+## `DEBUGGER__.open`
+
+After requiring `debug/session`, you can start a debug session using the following methods.
+They are convenient if you want to specify debug configurations with Ruby code.
+
+- `DEBUGGER__.open(**kw)`: opens a debug port using the specified configuration (by default opens a UNIX domain socket).
+- `DEBUGGER__.open_unix(**kw)`: opens a debug port through UNIX domain socket.
+- `DEBUGGER__.open_tcp(**kw)`: opens a debug port through TCP/IP.
+
+For example:
+
+```ruby
+require 'debug/session'
+DEBUGGER__.open
+
+# your application code
+```
+
+## UNIX Domain Socket (UDS)
+
+By default, the debugger uses UNIX domain socket for remote connection. You can customize the socket directory and socket path with environment variables:
+
+- `RUBY_DEBUG_SOCK_DIR` - This is where the debugger will create and look for socket files.
+- `RUBY_DEBUG_SOCK_PATH` - This is the socket file's absolute path that will be used to connect to the debugger.
+  - Currently, the debugger expects socket files to have a `ruby-debug-#{USER}` prefix during socket lookup. But this may be changed in the future.
```

Comment:
I know you don't want to maintain this, but it's important for anyone who wants to set up the debugger at scale. And I also make it clear that it's subject to change.